src/client.rs

```diff
@@ -280,3 +270,368 @@ where
         Ok(signed)
     }
 }
+
+/// This struct represents a subscription to the progress of some transaction, and is
+/// returned from [`SubmittableExtrinsic::sign_and_submit_then_watch()`].
+pub struct TransactionProgress<'client, T: Config> {
+    sub: Option<RpcSubscription<TransactionStatus<T::Hash, T::Hash>>>,
+    ext_hash: T::Hash,
+    client: &'client Client<T>,
+}
+
+impl<'client, T: Config> std::fmt::Debug for TransactionProgress<'client, T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TransactionProgress")
+            .field("sub", &self.sub)
```

Comment:
Does this actually print something useful?

Reply:
Not really; the `ext_hash` maybe at best. The `sub` debug is quite verbose and I haven't paid much attention to what it contains.